server/src/workflow-management/scheduler/index.ts

```
@@ -106,13 +119,15 @@ async function executeRun(id: string) {
     }
 
-    const currentPage = await browser.getCurrentPage();
+    let currentPage = await browser.getCurrentPage();
     if (!currentPage) {
       throw new Error('Could not create a new page');
     }
 
+    const workflow = AddGeneratedFlags(recording.recording);
     const interpretationInfo = await browser.interpreter.InterpretRecording(
-      recording.recording, currentPage, plainRun.interpreterSettings);
+      workflow, currentPage, (newPage: Page) => currentPage = newPage, plainRun.interpreterSettings
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)